### PR TITLE
Use the new accelerator DSL for tanzu-java-web-app

### PR DESCRIPTION
**Pull request by trisberg**

### CODE REVIEW

**Review by tyuzhikova-vmware — APPROVED** [review]

File: tanzu-java-web-app/accelerator.axl [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    // Include devcontainer files if selected
+    + InvokeFragment(reference: "devcontainer")
+  }
+  UniquePath(strategy: UseLast)
+  {
+    Include({'**'})
+    // Update workload.yaml files with Git repo if created
+    + InvokeFragment(reference: "tap-workload")
+  }
+  UniquePath(strategy: UseLast)
```
Comment (ericbottard) [inline tanzu-java-web-app/accelerator.axl:71]: All of those could be combined in a single merge like they used to

File: tanzu-java-web-app/accelerator.axl [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }
+  applyTo("catalog/*.yaml") {
+    ReplaceText({{text: "tanzu-java-web-app", with: #artifactId.toLowerCase()}})
+  }
+  applyTo("config/workload.yaml") {
+    ReplaceText({{text: "tanzu-java-web-app", with: #artifactId.toLowerCase()}})
+  }
+  applyTo("README.md") {
+    ReplaceText({{text: "tanzu-java-web-app", with: #artifactId.toLowerCase()}})
+  }
```
Comment (ericbottard) [inline tanzu-java-web-app/accelerator.axl:38]: All of those can be folded

File: tanzu-java-web-app/accelerator.axl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      Exclude({"Tiltfile", "Tiltfile_gradle_vscode"})
+      RewritePath(regex: "Tiltfile_gradle_intellij", rewriteTo: 'Tiltfile')
+    }
+    if (#ide == "vscode") {
+      Exclude({"Tiltfile", "Tiltfile_gradle_intellij"})
+      RewritePath(regex: "Tiltfile_gradle_vscode", rewriteTo: 'Tiltfile')
+    }
+    {
+      Exclude({"pom.xml"})
+      + InvokeFragment(reference: "build-wrapper-gradle")
```
Comment (ericbottard) [inline tanzu-java-web-app/accelerator.axl:22]: Not an action item for this PR, but if we're going to promote chain over merge going forward, our fragments would be better off touching only the files they care about and leaving the others as-is.
This 4 lines block would then just be
```
  InvokeFragment("build-wrapper-gradle")
```

File: tanzu-java-web-app/accelerator.axl (hunk 1 above)
Comment (trisberg) [reply tanzu-java-web-app/accelerator.axl:71]: I tried that but then I did not get the changes from java-version in the final output, maybe I did it wrong

File: tanzu-java-web-app/accelerator.axl (hunk 2 above)
Comment (trisberg) [reply tanzu-java-web-app/accelerator.axl:38]: indeed, let me do that

File: tanzu-java-web-app/accelerator.axl (hunk 3 above)
Comment (trisberg) [reply tanzu-java-web-app/accelerator.axl:22]: Not sure what we should promote in terms of chain over merge, but it seemed more natural to use a chain with the new DSL syntax. Throwing in a bunch of `+` to get a merge , to me, seems fine on smaller snippets but not at the top level.

**Pull request merged**